src/segments/nba.go

```diff
@@ -0,0 +1,257 @@
+	Time    string `json:"time"`
+	Code    int    `json:"code"`
+}
+
+type Scoreboard struct {
+	GameDate   string `json:"gameDate"`
+	LeagueID   string `json:"leagueId"`
+	LeagueName string `json:"leagueName"`
+	Games      []Game `json:"games"`
+}
+
+type Game struct {
+	GameID            string      `json:"gameId"`
+	GameCode          string      `json:"gameCode"`
+	GameStatus        int         `json:"gameStatus"`
+	GameStatusText    string      `json:"gameStatusText"`
+	Period            int         `json:"period"`
+	GameClock         string      `json:"gameClock"`
+	GameTimeUTC       string      `json:"gameTimeUTC"`
+	GameEt            string      `json:"gameEt"`
+	RegulationPeriods int         `json:"regulationPeriods"`
+	IfNecessary       bool        `json:"ifNecessary"`
+	SeriesGameNumber  string      `json:"seriesGameNumber"`
+	SeriesText        string      `json:"seriesText"`
+	SeriesConference  string      `json:"seriesConference"`
+	PoRoundDesc       string      `json:"poRoundDesc"`
+	GameSubtype       string      `json:"gameSubtype"`
+	HomeTeam          Team        `json:"homeTeam"`
+	AwayTeam          Team        `json:"awayTeam"`
+	GameLeaders       GameLeaders `json:"gameLeaders"`
+	PbOdds            PbOdds      `json:"pbOdds"`
+}
+
+type Team struct {
+	TeamID            int      `json:"teamId"`
+	TeamName          string   `json:"teamName"`
+	TeamCity          string   `json:"teamCity"`
+	TeamTricode       string   `json:"teamTricode"`
+	Wins              int      `json:"wins"`
+	Losses            int      `json:"losses"`
+	Score             int      `json:"score"`
+	Seed              *int     `json:"seed"`    // Pointer because it can be null
+	InBonus           *string  `json:"inBonus"` // Pointer because it can be null
+	TimeoutsRemaining int      `json:"timeoutsRemaining"`
+	Periods           []Period `json:"periods"`
+}
+
+type Period struct {
+	Period     int    `json:"period"`
+	PeriodType string `json:"periodType"`
+	Score      int    `json:"score"`
+}
+
+type GameLeaders struct {
+	HomeLeaders PlayerLeaders `json:"homeLeaders"`
+	AwayLeaders PlayerLeaders `json:"awayLeaders"`
+}
+
+type PlayerLeaders struct {
+	PersonID    int     `json:"personId"`
+	Name        string  `json:"name"`
+	JerseyNum   string  `json:"jerseyNum"`
+	Position    string  `json:"position"`
+	TeamTricode string  `json:"teamTricode"`
+	PlayerSlug  *string `json:"playerSlug"` // Pointer because it can be null
+	Points      int     `json:"points"`
+	Rebounds    int     `json:"rebounds"`
+	Assists     int     `json:"assists"`
+}
+
+type PbOdds struct {
+	Team      *string `json:"team"` // Pointer because it can be null
+	Odds      float64 `json:"odds"`
+	Suspended int     `json:"suspended"`
+}
+
+// NightscoutData struct contains the API data
+type NBAData struct {
+	HomeTeam       string
+	AwayTeam       string
+	Time           string
+	GameDate       string
+	StartTimeUTC   string
+	GameStatus     int // 1 = scheduled, 2 = in progress, 3 = finished
```

Comment:
If you make this into a custom type, you can add a `String()` function to get the representation.